src/main/java/Duke/parser/Parser.java

```diff
@@ -0,0 +1,40 @@
+package Duke.parser;
+
+import Duke.commands.*;
+import Duke.exceptions.DukeException;
+
+public class Parser {
```

Comment:
I like how you abstracted the functionalities here into separate Command classes